commands/submit.go

```diff
@@ -78,8 +78,8 @@ func submitReview(repo repository.Repo, args []string) error {
 	if err := repo.VerifyGitRef(target); err != nil {
 		return err
 	}
-	source, err := r.GetHeadCommit()
-	if err != nil {
+	source := r.Request.ReviewRef
```

Comment:
We still want to use the result of `GetHeadCommit` in the below check. It will actually use the review ref head if the metadata linking the review commit to the review ref head is properly set up.

However, we *also* want to give the user instructions on how to fix it if that metadata is not correct.

I think a better fix might be to add logic right after this (right before we check the ancestry of the target ref) to double check that the metadata is correct and give the user instructions on how to fix it if it is not.

Something like:

```go
	isReviewAssociatedWithRef, err := repo.IsAncestor(source, r.Request.ReviewRef)
	if err != nil {
		return err
	}
	if !isReviewAssociatedWithRef {
		return fmt.Errorf("The review commit is not associated with the review ref %q. If the change has been rebased, then reassociate the review commit with the review ref using the command `git appraise rebase %s`", r.Request.ReviewRef, r.Revision)
	}
```